pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -171,6 +171,21 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 
 //nolint:gocyclo
 func (c *Controller) finalize(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
+	// perform a no-op for whatever the status condition is currently set to
+	// so that we bump the observed generation to the latest and prevent the nodeclaim
+	// root status from entering an `Unknown` state
+	stored := nodeClaim.DeepCopy()
+	for _, condition := range nodeClaim.GetConditions() {
```

Comment:
I think we have to be careful here about the status conditions that we are updating -- we should just update the ones that are handled by the lifecycle controller -- no other ones

Reply:
We can fix with `nodeClaim.StatusConditions().List()`